proposals/provenance-logging.md

```diff
@@ -0,0 +1,128 @@
+## Motivation
+Reproducible ML is a [big problem](https://petewarden.com/2018/03/19/the-machine-learning-reproducibility-crisis/). 
+Kubeflow could add a lot of values to users by making it easy for users to track how datasets and models were produced.
```

Comment:
"In order to address requirements in key scenarios, Kubeflow must provide an easy way to track..."